### PR TITLE
Document cert-manager installation.

### PR DESCRIPTION
**Pull request by derpsteb**
### Proposed change(s)

- Explain that cert-manager is preinstalled and what limitations this brings in regards to user-managed cert-manager.

### Checklist

- [ ] Update [CHANGELOG.md](https://github.com/edgelesssys/constellation/blob/main/CHANGELOG.md)
- [x] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Link to Milestone

### CODE REVIEW

File: docs/docs/architecture/cert-manager.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Cert-manager
+
+Constellation ships with cert-manager preinstalled.
+The default installation is part of the `kube-system` namespace, as all other Constellation-managed components.
+You are free to install more instances of `cert-manager` into other namespaces.
+However, please be aware that any new installation need to use the same version as the one installed with Constellation.
```
Comment (datosh) [inline docs/docs/architecture/cert-manager.md:6]: We could use a `:::caution` or `:::info` box to highlight this information.

**Review by datosh — COMMENTED** [review]
This is currently not showing up on the docs and needs to be added to `constellation/docs/sidebars.js`

**Review by m1ghtym0 — COMMENTED** [review]
I'm unsure if `/architecture` is the best place to put it.
Makes sense ofc, but not sure if people would look into that.
We could also place it into `Workflows` like `Install cert-manager` or smth.
Or we have another top-level category, for Kubernetes operations for example.

File: docs/docs/architecture/cert-manager.md (hunk 1 above)
Comment (derpsteb) [reply docs/docs/architecture/cert-manager.md:6]: Couldn't find any information on this in the official docs. But I think there is nothing to be done apart from setting the namespace.

**Comment by derpsteb** [comment]
Moved to Workflows.

File: docs/docs/workflows/installing-cert-manager.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+Please read this section before installing cert-manager.
+:::
+Constellation ships with cert-manager preinstalled.
+The default installation is part of the `kube-system` namespace, as all other Constellation-managed components.
+You are free to install more instances of `cert-manager` into other namespaces.
+However, please be aware that any new installation need to use the same version as the one installed with Constellation.
+Or rely on the same CRD versions as Constellation's installation.
+Also remember to set the `installCRDs` value to `false` when installing new `cert-manager` instances.
+It will create problems if you have two installations of cert-manager depending on different versions of the installed CRDs.
+CRDs are cluster-wide resources and `cert-manager` uses depends on specific versions of those CRDs for each release.
```
Comment (datosh) [inline docs/docs/workflows/installing-cert-manager.md:12]:
```suggestion
CRDs are cluster-wide resources and `cert-manager` depends on specific versions of those CRDs for each release.
```

**Review by datosh — APPROVED** [review]
I think the word is extra, but other than that this LGTM

**Pull request merged**